spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/procedures/SetCurrentSnapshotProcedure.java

```diff
@@ -42,7 +43,8 @@ class SetCurrentSnapshotProcedure extends BaseProcedure {
   private static final ProcedureParameter[] PARAMETERS =
       new ProcedureParameter[] {
         ProcedureParameter.required("table", DataTypes.StringType),
-        ProcedureParameter.required("snapshot_id", DataTypes.LongType)
+        ProcedureParameter.optional("snapshot_id", DataTypes.LongType),
+        ProcedureParameter.optional("tag", DataTypes.StringType)
```

Comment:
maybe we'd want to use `ref` so that we could refer to a branch/tag?

Reply:
I don’t think “ref” is user facing and would rather add one more option for “branch”.

Reply:
We do call it `refs()` in `Table`, which is as user facing as it can be. We also have `refs` system table. I don't feel strongly here.

Reply:
@jackye1995, thoughts?

Reply:
If we're going to make it work with `branch`, I'd prefer `ref` or `snapshot_reference`.

Reply:
Update with option `ref` to support both `tag` and `branch`.